scripts/usb_scan.py

```diff
@@ -0,0 +1,545 @@
+            if p not in self:
+                return False
+        # TODO check a few important props(interface num, conf value) here
+        # try:
+        #     # usb_actconfig_attr(bNumInterfaces, "%2d\n");
+        #     except (KeyError, ValueError):
+        #     return False
+        return super(self.__class__, self).is_initialized()
+
+    def is_class_hub(self):
+        return self._is_class_hub(self._CLASS)
+
+    def add_interface(self, interface):
+        if interface in self.interfaces:
+            Log().debug("overriding existing interface: " + interface)
+            self.interfaces.remove(interface)
+        self.interfaces.add(interface)
+
+    def del_interface(self, interface):
+        if interface in self.interfaces:
+            self.interfaces.remove(interface)
+        else:
+            Log().debug("try to remove unknown interface: " + interface)
+
+    def get_all_interfaces(self):
+        return self.interfaces
+
+    def is_ready(self):
+        n = int(self[self._NUM_INTERFACES])
+        return n > 0 and n == len(self.interfaces)
+
+    def is_allow(self):
+        return self.allow
+
+    def set_allow(self, allow):
+        self.allow = allow
+
+    def get_all_storages(self):
+        ret = []
+        for i in self.interfaces:
+            ret.extend(i.get_storage_list())
+        return sorted(set(ret))
+
+    def is_child_of(self, parent):
+        # USB device is at the very top, since we don't care PCI info
+        return False
+
+
+class UsbInterface(UsbObject):
+    _NUMBER = "bInterfaceNumber"
+    _CLASS = "bInterfaceClass"
+    _SUB_CLASS = "bInterfaceSubClass"
+    _PROTOCOL = "bInterfaceProtocol"
+
+    _PROPS = [_NUMBER, _CLASS, _SUB_CLASS, _PROTOCOL]
+
+    def __init__(self, node, props):
+        super(UsbInterface, self).__init__(node)
+        for p in self._PROPS:
+            if p in props:
+                self[p] = props[p]
+        self.storages = set()
+
+    def __str__(self):
+        st = super(UsbInterface, self).__str__()
+        for s in self.storages:
+            st += "\n\t " + str(s)
+        return st
+
+    def is_class_hub(self):
+        return self._is_class_hub(self._CLASS)
+
+    def is_initialized(self):
+        for p in self._PROPS:
+            if p not in self:
+                return False
+        # TODO check a few important props(class, subclass, protocol) here
+        return super(self.__class__, self).is_initialized()
+
+    def get_storages(self):
+        return self.storages
+
+    def add_storage(self, storage):
+        if storage in self.storages:
+            Log().debug("overriding existing storage: " + storage)
+            self.storages.remove(storage)
+        self.storages.add(storage)
+
+    def del_storage(self, storage):
+        if storage in self.storages:
+            self.storages.remove(storage)
+        else:
+            Log().debug("try to remove unknown storage: " + storage)
+
+    # def get_device_node(self):
```

Comment:
If this code is not required please delete it.

Reply:
done.